### DISCUSSION TITLE
Which GPT model is used by `copilot.lua`

### DISCUSSION THREAD

**Discussion by pidgeon777** [zbirenbaum/copilot.lua#301] [General]
I am writing to inquire about the specific version of the Copilot model that is being utilized by **copilot.lua**. The existence of multiple versions of the Copilot model, as depicted in the image below, has piqued my interest.

![Copilot Model Versions](https://github.com/user-attachments/assets/5e62f5df-c912-4e07-ab55-d2b5817bbb13)

In addition to this, I am also interested in knowing whether **copilot.vim** uses the same version of the model as **copilot.lua**.

Furthermore, I would like to know if there is a method to dynamically determine the version of the model being used, possibly through a command.

Lastly, I am curious to know if it is possible for the user to select a different model version for use.

**Comment by LorenzoBettini** [comment]
I'd be interested to.

I seem to understand that this is not documented at all?

**Reply by AntoineGS** [reply]
There are currently no options available (server-side), though maintainers of the official Copilot repos have mentioned wanting to add an API call to list available models at some point, not sure if it will ever happen 🤷 time will tell.

**Reply by LorenzoBettini** [reply]
So, which model is used? ChatGPT?

**Reply by AntoineGS** [reply]
Yeah it's currently GPT4o

**Comment by jinzhongjia** [comment]
now,gpt-41-copilot is available

**Reply by pidgeon777** [reply]
> now,gpt-41-copilot is available

You mean, as autocompletion model?

**Reply by jinzhongjia** [reply]
yes, the report is here https://github.blog/changelog/2025-08-27-copilot-code-completion-now-uses-the-gpt-4-1-copilot-model/

**Reply by pidgeon777** [reply]
> yes, the report is here https://github.blog/changelog/2025-08-27-copilot-code-completion-now-uses-the-gpt-4-1-copilot-model/

This is great, 4.1 has a larger context window, hopefully there will be good improvements during autocompletion. Do you happen to know if it is already set as base model in *copilot.lua*?

**Reply by jinzhongjia** [reply]
I didn't know, but I set it  manully

**Reply by pidgeon777** [reply]
> I didn't know, but I set it  manully

How?

**Reply by jinzhongjia** [reply]
may be more useful

**Reply by pidgeon777** [reply]
> may be more useful

Yes I mean, how did you manually set the chosen autocompletion model?

It would have a great info to know 🙂.

**Reply by AntoineGS** [reply]
Pretty sure the LSP release I merged yesterday includes it, someone could ask on the [official repo](https://github.com/github/copilot-language-server-release/).
If I request an invalid model from the LSP it gives this error:
```
2025-08-28 10:06:05.758 [45] [ERROR]: LSP message: [default] User selected model gpt-44-copilot is not in the list of generic models: gpt-41-copilot, falling back to default model.
```
Since there is only one model I reckon it is the default.

**Comment by ChenZhongPu** [comment]
How to set the default model?

**Reply by pidgeon777** [reply]
> How to set the default model?

Is it possible to choose a model?
